autoemulate/experimental/tuner.py

```diff
@@ -57,15 +59,31 @@ def run(self, model_class: type[Emulator]) -> tuple[list[float], list[ModelConfi
             }
             if isinstance(model_class, gpytorch.models.ExactGP):
                 m = model_class(train_x, train_y, **params_sample)
+                assert isinstance(m, Emulator)
                 m.fit(train_x, train_y)
             else:
                 m = model_class(**params_sample)
+                assert isinstance(m, Emulator)
                 # TODO: check if pass as dataloader
                 m.fit(train_x, train_y)
 
             # evaluate
             y_pred = m.predict(val_x)
-            score = self.score_f(val_y, y_pred.detach().numpy())
+            # handle types
+            if isinstance(y_pred, TensorLike):
+                score = self.score_f(val_y, y_pred.detach().numpy())
+            elif isinstance(y_pred, DistributionLike):
+                score = self.score_f(val_y, y_pred.mean.detach().numpy())
+            elif (
```

Comment:
when is a tuple returned?

Reply:
I think this is for cases when we have any emulators that returns parameters of distribution (e.g. mean and std in [sklearn's GP](https://scikit-learn.org/stable/modules/generated/sklearn.gaussian_process.GaussianProcessRegressor.html#sklearn.gaussian_process.GaussianProcessRegressor.predict)). We might consider always returning a `DsitributionLike` from emulators as part of the refactor? (relates to #327)

Reply:
Makes sense, thanks! Since we already have an issue for this let's leave it as is.